scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -631,21 +622,12 @@ def apply_filters(self, **kwargs):
             instance_list = []
             for plat in platform_scope:
                 instance = TestInstance(ts, plat, self.env.outdir)
-                if runnable:
-                    tfilter = 'runnable'
-                else:
-                    tfilter = 'buildable'
-
-                instance.run = instance.check_runnable(
-                    self.options.enable_slow,
-                    tfilter,
-                    self.options.fixture
-                )
-                if runnable and self.hwm.duts:
-                    for h in self.hwm.duts:
-                        if h.platform == plat.name:
-                            if ts.harness_config.get('fixture') in h.fixtures:
-                                instance.run = True
+                # If instances were inherited (existing testplan used) don't add new nor evaluate filtered instances
+                if instances_were_inherited:
+                    if instance.name not in self.instances:
+                        continue
+                    elif self.instances[instance.name].status == "filtered":
+                        continue
```

Comment:
`run_id` of new created `TestInstance` should be inherited from old one. For example:

```python
else:
    instance.run_id = self.instances[instance.name].run_id
```

Reply:
Fixed, thanks!